src/spxmod/space.py

```diff
@@ -115,7 +114,10 @@ def encode(self, data: DataFrame, column: str = "intercept") -> DataFrame:
                 .eval("index")
                 .to_numpy()
             )
-        return coo_matrix((val, (row, col)), shape=(len(data), self.size))
+        index = ~np.isnan(col)
```

Comment:
will there be `nan` `col`? the number of rows of the design matrix should be the same with the length of data.

Reply:
Yes, if we've applied the filter, then `self.span` won't have all of the IDs contained in data, so when we do the left merge there will be nans in the `index` column. This causes an error with `coo_matrix` unless we remove the nan entries. The number of rows will still be the length of the data since `shape=(len(data), self.size)`.